2_run/src/run_glm3.R

```diff
@@ -52,11 +52,23 @@ extract_glm_output <- function(nc_filepath, nml_obj, export_fl) {
   lake_depth <- glmtools::get_nml_value(nml_obj, arg_name = 'lake_depth')
   export_depths <- seq(0, lake_depth, by = 0.5)
   temp_data <- glmtools::get_temp(nc_filepath, reference = 'surface', z_out = export_depths) %>%
-    mutate(date = as.Date(lubridate::floor_date(DateTime, 'days'))) %>% select(-DateTime)
-  glmtools::get_var(nc_filepath, var_name = 'hice') %>%
-    dplyr::mutate(ice = hice > 0, date = as.Date(lubridate::ceiling_date(DateTime, ' days'))) %>% dplyr::select(-hice, -DateTime) %>%
-    dplyr::left_join(temp_data, ., by = 'date') %>%
-    select(time = date, everything()) %>%
+    mutate(time = as.Date(lubridate::floor_date(DateTime, 'days'))) %>% 
+    select(-DateTime)
+  layers_data <- glmtools::get_var(nc_filepath, 'NS') %>%
+    rename(n_layers = NS) %>%
+    mutate(time = as.Date(lubridate::ceiling_date(DateTime, ' days'))) %>%
+    select(-DateTime)
+  evap_data <- glmtools::get_var(nc_filepath, 'evap') %>%
+    mutate(time = as.Date(lubridate::ceiling_date(DateTime, ' days'))) %>%
+    select(-DateTime)
+  ice_data <- glmtools::get_var(nc_filepath, var_name = 'hice') %>%
+    mutate(ice = hice > 0, time = as.Date(lubridate::ceiling_date(DateTime, ' days'))) %>%
+    select(-DateTime) 
+  all_results <- layers_data %>%
+    left_join(evap_data, ., by = 'time') %>%
+    left_join(ice_data, ., by = 'time') %>%
+    left_join(temp_data, ., by = 'time') %>%
```

Comment:
instead of `left_join(data, ., by = 'time')`, you could follow the pattern `right_join(evap_data, by = 'time')`. Or reorder so that you don't have to keep specifying `.` over and over. I think re-ordering would be the most common pattern and more readable

Reply:
Sorry that was remnant code from some earlier copy and paste that I just hadn't edited. Fixed now.